[CDAP-16959] Avoiding creation of uuid if already present for macros

itsanudeep opened the pull request:
Re-using existing UUID when retrieving macros for display.

JIRA: https://issues.cask.co/browse/CDAP-16959

itsanudeep committed:
Avoiding creation of uuid if already present for macros

ajainarayanan commented:
<img width="871" alt="Screen Shot 2020-06-15 at 8 51 33 AM" src="https://user-images.githubusercontent.com/1452845/84679862-b1632f80-aee6-11ea-816e-651b6404e97c.png">
Seems like we removed the delete button from runtime arguments in preview

ajainarayanan commented:
When there is a unfilled macro during preview, I am unable to stop the preview 
![unable_to_stop1](https://user-images.githubusercontent.com/1452845/84679926-c770f000-aee6-11ea-8d9a-5e40d24f685c.gif)

ajainarayanan commented on cdap-ui/app/cdap/components/PipelineConfigurations/Store/index.js:
```diff
@@ -197,8 +197,8 @@ const getRuntimeArgsForDisplay = (currentRuntimeArgs, macrosMap) => {
-      if (currentPair.notDeletable && currentPair.provided) {
-        providedMacros[key] = currentPair.value;
+      if (currentPair.uniqueId && Object.prototype.hasOwnProperty.call(macrosMap, key)) {
+        macrosWithIds[key] = currentPair.uniqueId;
```
Can't we combine this and `const macros = ` statement into one? Do we have to reset `currentRuntimeArgs.pairs` 3 times?

itsanudeep replied:
Simplified the implementation here. Did not change the implementation in hydrator-service as that needs more work and reusing this implementation won't work there.

itsanudeep commented:
> <img alt="Screen Shot 2020-06-15 at 8 51 33 AM" width="871" src="https://user-images.githubusercontent.com/1452845/84679862-b1632f80-aee6-11ea-816e-651b6404e97c.png">
> 
> Seems like we removed the delete button from runtime arguments in preview

The delete button was hidden to the right, had to horizontally scroll. Fixed it.

itsanudeep commented:
> When there is a unfilled macro during preview, I am unable to stop the preview
> ![unable_to_stop1](https://user-images.githubusercontent.com/1452845/84679926-c770f000-aee6-11ea-8d9a-5e40d24f685c.gif)

This behavior exists in develop as well. https://issues.cask.co/browse/CDAP-16972 to fix this.

itsanudeep commented:
BUILD: https://builds.cask.co/browse/CDAP-URUT275-1
